src/controller/CHIPDeviceController.cpp

```diff
@@ -1190,6 +1209,170 @@ void DeviceCommissioner::OnSessionEstablished()
     }
 }
 
+CHIP_ERROR DeviceCommissioner::SendCertificateChainRequestCommand(Device * device, CertificateChainType certificateChainType)
+{
+    ChipLogDetail(Controller, "Sending Certificate Chain request to %p device", device);
+    VerifyOrReturnError(device != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    chip::Controller::OperationalCredentialsCluster cluster;
+    cluster.Associate(device, 0);
+
+    mCertificateChainBeingRequested = certificateChainType;
+
+    Callback::Cancelable * successCallback = mCertChainResponseCallback.Cancel();
+    Callback::Cancelable * failureCallback = mOnCertChainFailureCallback.Cancel();
+
+    ReturnErrorOnFailure(cluster.CertChainRequest(successCallback, failureCallback, certificateChainType));
+    ChipLogDetail(Controller, "Sent Certificate Chain request, waiting for the DAC Certificate");
+    return CHIP_NO_ERROR;
+}
+
+void DeviceCommissioner::OnCertChainFailureResponse(void * context, uint8_t status)
+{
+    ChipLogProgress(Controller, "Device failed to receive the Certificate Chain request Response: 0x%02x", status);
+    DeviceCommissioner * commissioner = reinterpret_cast<DeviceCommissioner *>(context);
+    commissioner->mCertChainResponseCallback.Cancel();
+    commissioner->mOnCertChainFailureCallback.Cancel();
+    // TODO: Map error status to correct error code
+    commissioner->OnSessionEstablishmentError(CHIP_ERROR_INTERNAL);
```

Comment:
This is actually not a Session Establishment error...

@pan-apple @cecille  Is there a better way to error-out on commissioning failures?